modules/reactor/src/meshgenerators/PatternedCartesianMeshGenerator.C

```diff
@@ -222,6 +235,35 @@ PatternedCartesianMeshGenerator::PatternedCartesianMeshGenerator(const InputPara
                  "This parameter and background_block_name must not be set when the "
                  "pattern_boundary is none.");
   }
+  if (_use_reporting_id)
+  {
+    if (_use_exclude_id && _assign_type != "cell")
+      paramError("exclude_id", "works only when \"assign_type\" is equal 'cell'");
+    if (!isParamValid("id_pattern") && _assign_type == "manual")
+      paramError("id_pattern", "required when \"assign_type\" is equal to 'manual'");
+
+    if (_assign_type == "manual")
+      _id_pattern = getParam<std::vector<std::vector<dof_id_type>>>("id_pattern");
+    _exclude_ids.resize(_input_names.size());
+    if (_use_exclude_id)
+    {
+      std::vector<MeshGeneratorName> exclude_id_name =
+          getParam<std::vector<MeshGeneratorName>>("exclude_id");
+      for (unsigned int i = 0; i < _input_names.size(); ++i)
+      {
+        _exclude_ids[i] = false;
+        for (auto input_name : exclude_id_name)
+          if (_input_names[i] == input_name)
+          {
+            _exclude_ids[i] = true;
+            break;
+          }
+      }
+    }
+    else
+      for (unsigned int i = 0; i < _input_names.size(); ++i)
+        _exclude_ids[i] = false;
+  }
```

Comment:
add some comments please. I know it s trivial for us right now, but it helps later